// slackin/node/splash.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = splash;

var _vd = require('vd');

var _vd2 = _interopRequireDefault(_vd);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function splash(_ref) {
  var path = _ref.path;
  var name = _ref.name;
  var org = _ref.org;
  var coc = _ref.coc;
  var logo = _ref.logo;
  var active = _ref.active;
  var total = _ref.total;
  var channels = _ref.channels;
  var iframe = _ref.iframe;

  var div = (0, _vd2.default)('.splash', !iframe && (0, _vd2.default)('.logos', logo && (0, _vd2.default)('.logo.org'), (0, _vd2.default)('.logo.slack')), (0, _vd2.default)('p', 'Join ', (0, _vd2.default)('b', name),
  // mention single single-channel inline
  channels && channels.length === 1 && (0, _vd2.default)('span', ' #', channels[0]), ' on Slack.'), (0, _vd2.default)('p.status', active ? [(0, _vd2.default)('b.active', active), ' users online now of ', (0, _vd2.default)('b.total', total), ' registered.'] : [(0, _vd2.default)('b.total', total), ' users are registered so far.']), (0, _vd2.default)('form',
  // channel selection when there are multiple
  channels && channels.length > 1 && (0, _vd2.default)('select.form-item name=channel', channels.map(function (channel) {
    return (0, _vd2.default)('option', { value: channel, text: channel });
  })), (0, _vd2.default)('input.form-item type=email placeholder=you@yourdomain.com ' + (!iframe ? 'autofocus' : '')), coc && (0, _vd2.default)('.coc', (0, _vd2.default)('label', (0, _vd2.default)('input type=checkbox name=coc value=1'), 'I agree to the ', (0, _vd2.default)('a', { href: coc, target: '_blank' }, 'Code of Conduct'), '.')), (0, _vd2.default)('button.loading', 'Get my Invite')), !iframe && (0, _vd2.default)('p.signin', 'or ', (0, _vd2.default)('a href=https://' + org + '.slack.com target=_top', 'sign in'), '.'), !iframe && (0, _vd2.default)('footer', 'powered by ', (0, _vd2.default)('a href=http://rauchg.com/slackin target=_blank', 'slackin')), style({ logo: logo, active: active, iframe: iframe }),
  // xxx: single build
  (0, _vd2.default)('script', '\n      data = {};\n      data.path = ' + JSON.stringify(path) + ';\n    '), (0, _vd2.default)('script src=https://cdn.socket.io/socket.io-1.4.4.js'), (0, _vd2.default)('script src=' + path + 'assets/superagent.js'), (0, _vd2.default)('script src=' + path + 'assets/client.js'));
  return div;
}

var pink = '#E01563';

function style() {
  var _ref2 = arguments.length <= 0 || arguments[0] === undefined ? {} : arguments[0];

  var logo = _ref2.logo;
  var active = _ref2.active;
  var iframe = _ref2.iframe;

  var css = _vd2.default.style();

  css.add('.splash', {
    'width': iframe ? '250px' : '300px',
    'margin': iframe ? '0' : '200px auto',
    'text-align': 'center',
    'font-family': '"Helvetica Neue", Helvetica, Arial'
  });

  if (iframe) {
    css.add('body, html', {
      'margin': '0',
      'padding': '0',
      'background': '#FAFAFA',
      'overflow': 'hidden' // ff
    });

    css.add('.splash', {
      'box-sizing': 'border-box',
      'padding': '10px'
    });
  }

  if (!iframe) {
    css.media('(max-width: 500px)').add('.splash', {
      'margin-top': '100px'
    });
  }

  css.add('.head', {
    'margin-bottom': '40px'
  });

  css.add('.logos', {
    'position': 'relative',
    'margin-bottom': '40px'
  });

  if (!iframe) {
    css.add('.logo', {
      'width': '48px',
      'height': '48px',
      'display': 'inline-block',
      'background-size': 'cover'
    });

    css.add('.logo.slack', {
      'background-image': 'url(assets/slack.svg)'
    });

    if (logo) {
      var pw = 10; // '+' width
      var lp = 30; // logos separation

      css.add('.logo.org::after', {
        'position': 'absolute',
        'display': 'block',
        'content': '"+"',
        'top': '15px',
        'left': '0',
        'width': '300px',
        'text-align': 'center',
        'color': '#D6D6D6',
        'font': '15px Helvetica Neue'
      });

      css.add('.logo.org', {
        'background-image': 'url(' + logo + ')',
        'margin-right': lp + pw + lp + 'px'
      });
    }
  }

  css.add('.coc', {
    'font-size': '12px',
    padding: '15px 0 5px',
    color: '#666'
  });

  if (iframe) {
    css.add('.coc', {
      'font-size': '11px',
      'padding-top': '10px'
    });

    css.add('.coc input', {
      position: 'relative',
      top: '-2px'
    });
  }

  css.add('.coc label', {
    cursor: 'pointer'
  });

  css.add('.coc input', {
    'appearance': 'none',
    '-webkit-appearance': 'none',
    border: 'none',
    'vertical-align': 'middle',
    margin: '0 5px 0 0'
  });

  css.add('.coc input::after', {
    content: '""',
    display: 'inline-block',
    width: '15px',
    height: '15px',
    'vertical-align': 'middle',
    background: 'url(/assets/checkbox.svg)',
    cursor: 'pointer'
  });

  css.add('.coc input:checked::after', {
    'background-position': 'right'
  });

  css.add('.coc a', {
    color: '#666'
  });

  css.add('.coc a:hover', {
    'background-color': '#666',
    'text-decoration': 'none',
    color: '#fff'
  });

  css.add('p', {
    'font-size': iframe ? '12px' : '15px',
    'margin': iframe ? '0 0 5px' : '5px 0'
  });

  if (iframe) {
    css.add('p.status', {
      'font-size': '11px'
    });
  }

  css.add('select', {
    'background': 'none'
  });

  css.add('button, .form-item', {
    'font-size': '12px',
    'margin-top': iframe ? '5px' : '10px',
    'vertical-align': 'middle',
    'display': 'block',
    'text-align': 'center',
    'box-sizing': 'border-box',
    'width': '100%',
    'padding': '9px'
  });

  css.add('button', {
    'color': '#fff',
    'font-weight': 'bold',
    'border-width': 0,
    'background': pink,
    'text-transform': 'uppercase',
    'cursor': 'pointer',
    'appearence': 'none',
    '-webkit-appearence': 'none',
    'outline': '0',
    'transition': 'background-color 150ms ease-in, color 150ms ease-in'
  });

  css.add('button.loading', {
    'pointer-events': 'none'
  });

  css.add('button:disabled', {
    'color': '#9B9B9B',
    'background-color': '#D6D6D6',
    'cursor': 'default',
    'pointer-events': 'none'
  });

  css.add('button.error', {
    'background-color': '#F4001E',
    'text-transform': 'none'
  });

  css.add('button.success:disabled', {
    'color': '#fff',
    'background-color': '#68C200'
  });

  css.add('button:not(.disabled):active', {
    'background-color': '#7A002F'
  });

  css.add('b', {
    'transition': 'transform 150ms ease-in'
  });

  css.add('b.grow', {
    'transform': 'scale(1.3)'
  });

  css.add('form', {
    'margin-top': iframe ? '10px' : '20px',
    'margin-bottom': '0'
  });

  css.add('input', {
    'color': '#9B9B9B',
    'border': '1px solid #D6D6D6'
  });

  if (iframe) {
    css.add('button, .form-item', {
      'height': '28px',
      'line-height': '28px',
      'padding': 0,
      'font-size': '11px'
    });
  }

  css.add('input:focus', {
    'color': '#666',
    'border-color': '#999',
    'outline': '0'
  });

  if (active) {
    css.add('.active', {
      'color': pink
    });
  }

  if (!iframe) {
    css.add('p.signin', {
      'padding': '10px 0 10px',
      'font-size': '11px'
    });

    css.add('p.signin a', {
      'color': pink,
      'text-decoration': 'none'
    });

    css.add('p.signin a:hover', {
      'background-color': '#E01563',
      color: '#fff'
    });
  }

  if (!iframe) {
    css.add('footer', {
      'color': '#D6D6D6',
      'font-size': '11px',
      'margin': '200px auto 0',
      'width': '300px',
      'text-align': 'center'
    });

    css.add('footer a', {
      'color': '#9B9B9B',
      'text-decoration': 'none',
      'border-bottom': '1px solid #9B9B9B'
    });

    css.add('footer a:hover', {
      'color': '#fff',
      'background-color': '#9B9B9B'
    });
  }

  return css;
}
